### PR TITLE
basic honkbots

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
this refactors honkbots into basic mobs. its mostly a faithful 1:1 refactor but i couldnt keep my hands to myselves so i gave them some new behaviors. 

honkbots now love playing with clowns, they will go seek out for clowns and celebrate around them. also, if the honkbot finds a banana peel or a slippery item near it, it will actively drag people onto them

honkbots will now go out of theirway to mess with secbots and annoy them

## Why It's Good For The Game
refactors hinkbots into basic bots and also undoes some of the silliness i did in the previous basic bot prs. i also added lazylist support to remove_thing_from_list.

## Changelog
:cl:
refactor: honkbots are now basic mobs, please report any bugs
add: honkbots will try to slip people on banana peels
/:cl:

### CODE REVIEW

**Comment by DrAmazing343** [comment]
ANOTHER BANGER BEN10O PR. HE NEVER FUCKING MISSES.

File: code/__DEFINES/dcs/signals/signals_mob/signals_mob_ai.dm [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /// Signal sent when a blackboard key is set to a new value
 #define COMSIG_AI_BLACKBOARD_KEY_SET(blackboard_key) "ai_blackboard_key_set_[blackboard_key]"
 
+///Signal sent before a blackboard key is cleared
+#define COMSIG_AI_BLACKBOARD_KEY_PRECLEAR(blackboard_key) "ai_blackboard_key_pre_clear_[blackboard_key]"
+	///prevent key clearing
+	#define COMSIG_DONT_CLEAR (1<<0)
```
Comment (MrMelbert) [inline code/__DEFINES/dcs/signals/signals_mob/signals_mob_ai.dm:7]: I do not think we should be allowing things to block blackboard clearing.

File: code/datums/ai/_ai_controller.dm [hunk 2]
```diff
@@ -604,8 +606,10 @@ multiple modular subtrees with behaviors
  * * key - A blackboard key
  * * thing - a value to set the blackboard key to.
  */
-/datum/ai_controller/proc/remove_thing_from_blackboard_key(key, thing)
+/datum/ai_controller/proc/remove_thing_from_blackboard_key(key, thing, lazylist = FALSE)
```
Comment (MrMelbert) [inline code/datums/ai/_ai_controller.dm:609]: Can I ask why you added lazylist support to this? It seems rather volatile for the nature of blackboard keys

Comment (MrMelbert) [reply code/datums/ai/_ai_controller.dm:609]: Wait, we have `set_blackboard_key_assoc_lazylist`. I forgot. Can you make this a separate proc instead like that?

File: code/modules/mob/living/basic/bots/cleanbot/cleanbot_ai.dm [hunk 3]
```diff
@@ -77,11 +76,13 @@
...
 			break
-		if(LAZYACCESS(ignore_list, REF(found_item)))
+		if(LAZYACCESS(ignore_list, found_item))
 			continue
+		if(get_turf(found_item) == get_turf(controller.pawn))
+			return found_item
 		var/list/path = get_path_to(controller.pawn, found_item, max_distance = BOT_CLEAN_PATH_LIMIT, access = controller.get_access())
 		if(!length(path))
-			controller.set_blackboard_key_assoc_lazylist(BB_TEMPORARY_IGNORE_LIST, REF(found_item), TRUE)
+			controller.set_blackboard_key_assoc_lazylist(BB_TEMPORARY_IGNORE_LIST, found_item, TRUE)
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/bots/cleanbot/cleanbot_ai.dm:85]: Why was this changed from `ref`s anyways? I thought you have a use for it but I don't see one

Comment (Ben10Omintrix) [reply code/modules/mob/living/basic/bots/cleanbot/cleanbot_ai.dm:85]: i initially used refs because i was worried about hard deletes, but upon studying the code i discovered that set_blackboard_key_assoc_lazylist handles the cleanup through signals so i figured it was unnecessary to use refs which are a bit more expensive to make

File: code/datums/ai/_ai_controller.dm (hunk 2 above)
Comment (MrMelbert) [reply code/datums/ai/_ai_controller.dm:609]: image

1 year ago me was on point fr

File: code/modules/mob/living/basic/bots/cleanbot/cleanbot_ai.dm (hunk 3 above)
Comment (MrMelbert) [reply code/modules/mob/living/basic/bots/cleanbot/cleanbot_ai.dm:85]: Fair enough

**Review by L0pz — APPROVED** [review]
MERGE!!!!!!

File: code/datums/ai/_ai_controller.dm (hunk 2 above)
Comment (Ben10Omintrix) [reply code/datums/ai/_ai_controller.dm:609]: alright but it might end up looking a bit copy-pasty

**Comment by Ben10Omintrix** [comment]
this pr is ready but i suggest its tmed first

**Comment by ZephyrTFA** [comment]
fuck it we ball
